src/results/DownloadContainer.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import { withRouter, RouteComponentProps } from 'react-router-dom';
+import idx from 'idx';
+import { RootState } from '../state/state-types';
+import DownloadView from './DownloadView';
+import { Column } from '../model/types/ColumnTypes';
+import { FileFormat, fileFormatToContentType } from './types/resultsTypes';
+import { getDownloadUrl } from '../utils/apiUrls';
+import fetchData from '../utils/fetchData';
+
+export const replaceExcelWithTsv = (fileFormat: FileFormat) =>
+  fileFormat === FileFormat.excel ? FileFormat.tsv : fileFormat;
+
+export const compareDownloadUrlsDisregardSize = (
+  url1: string,
+  url2: string
+) => {
+  const reSize = /&size=\d+/;
+  return url1.replace(reSize, '') === url2.replace(reSize, '');
+};
+
+type DownloadTableProps = {
+  tableColumns: Column[];
+  totalNumberResults: number;
+} & RouteComponentProps;
+
+const Download: React.FC<DownloadTableProps> = ({
+  tableColumns,
+  totalNumberResults,
+  history,
+  location: {
+    state: {
+      query,
+      selectedFacets,
+      sortColumn,
+      sortDirection,
+      selectedEntries,
+    },
+  },
+}) => {
+  const [selectedColumns, setSelectedColumns] = useState(tableColumns);
```

Comment:
That's a good question. I passed in the table columns because I assumed that a user would want to see download what they're seeing (assuming they're in table mode). If a user is using card mode and they want to download the results it's not clear which columns exactly they want to download so it would make sense to store these in a separate place as you've suggested. We might even need to know which view the user was using when showing the download columns. Shall we poll the team for their perspectives?

Reply:
Would you mind making a Jira about this and assigning it to Sangya?

Reply:
done: https://www.ebi.ac.uk/panda/jira/browse/TRM-23576